### PR TITLE
fix: 1292 Provide correct rules for price operation

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

This bug fix will make sure we differentiate and provide the correct rules for charge information operation and charge price operation, respectively.

## References

* #1292

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs [hunk 1]
```diff
@@ -29,29 +29,47 @@ public IValidationRuleSet CreateRules(ChargeOperationDto operation)
...
                 CreateRuleContainer(new ChargeIdRequiredValidationRule(chargeOperationDto), chargeOperationDto),
-                CreateRuleContainer(new ChargeNameHasMaximumLengthRule(chargeOperationDto), chargeOperationDto),
                 CreateRuleContainer(new ChargeOperationIdRequiredRule(chargeOperationDto), chargeOperationDto),
                 CreateRuleContainer(new ChargeOwnerIsRequiredValidationRule(chargeOperationDto), chargeOperationDto),
-                CreateRuleContainer(new ChargePriceMaximumDigitsAndDecimalsRule(chargeOperationDto), chargeOperationDto),
                 CreateRuleContainer(new ChargeTypeIsKnownValidationRule(chargeOperationDto), chargeOperationDto),
+                CreateRuleContainer(new StartDateTimeRequiredValidationRule(chargeOperationDto), chargeOperationDto),
+            };
+
+            rules.AddRange(chargeOperationDto.Points.Any()
```
Comment (x-platformcoder) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs:44]: Hvis vi læser en D18 ind med points, får vi så ikke valideret forkert? altså indtil vi har sikret os at vi ikke læser points ind i chargeoperations ved en D18 i cim konverteringen?

**Review by prtandrup — APPROVED** [review]
🌮

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs (hunk 1 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs:44]: He has a point 👀 D18 with points in a create operation will bypass input validation.

Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/InputValidation/Factories/ChargeOperationInputValidationRulesFactory.cs:44]: I don't see how we can change that with a quick fix. So maybe we just need to let it pass? and then as soon as we no longer read points for a D18 request the problem will "fix itself"?

**Pull request merged**